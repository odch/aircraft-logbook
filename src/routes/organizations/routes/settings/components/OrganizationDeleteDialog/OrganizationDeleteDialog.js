import React from 'react'
import PropTypes from 'prop-types'
import { injectIntl, intlShape, FormattedMessage } from 'react-intl'
import Button from '@material-ui/core/Button'
import TextField from '@material-ui/core/TextField'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogTitle from '@material-ui/core/DialogTitle'
import DeleteButton from '../../../../../../components/DeleteButton'

class OrganizationDeleteDialog extends React.Component {
  state = {
    id: ''
  }

  handleExited = () => {
    this.setState({
      id: ''
    })
  }

  handleIdChange = e => {
    this.setState({
      id: e.target.value
    })
  }

  handleSubmit = e => {
    e.preventDefault()
    if (this.props.onConfirm) {
      this.props.onConfirm()
    }
  }

  render() {
    return (
      <Dialog
        open={this.props.open}
        onClose={this.props.onClose}
        onExited={this.handleExited}
      >
        <DialogTitle>
          <FormattedMessage id="organizations.delete.dialog.title" />
        </DialogTitle>
        <form onSubmit={this.handleSubmit}>
          <DialogContent>
            <DialogContentText>
              <FormattedMessage id="organizations.delete.dialog.text" />
              &nbsp;
              <strong>{this.props.organizationId}</strong>
            </DialogContentText>
            <TextField
              autoFocus
              margin="dense"
              id="name"
              placeholder={this.props.intl.formatMessage({
                id: 'organizations.delete.dialog.id'
              })}
              type="text"
              fullWidth
              required
              value={this.state.id}
              onChange={this.handleIdChange}
            />
          </DialogContent>
          <DialogActions>
            <Button onClick={this.props.onClose}>
              <FormattedMessage id="organizations.delete.dialog.buttons.cancel" />
            </Button>
            <DeleteButton
              type="submit"
              color="secondary"
              variant="contained"
              disabled={this.state.id !== this.props.organizationId}
              label={this.props.intl.formatMessage({
                id: 'organizations.delete.dialog.buttons.delete'
              })}
            />
          </DialogActions>
        </form>
      </Dialog>
    )
  }
}

OrganizationDeleteDialog.propTypes = {
  organizationId: PropTypes.string.isRequired,
  open: PropTypes.bool,
  onConfirm: PropTypes.func,
  onClose: PropTypes.func,
  intl: intlShape
}

export default injectIntl(OrganizationDeleteDialog)
