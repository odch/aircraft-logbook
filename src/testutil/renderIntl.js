import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
