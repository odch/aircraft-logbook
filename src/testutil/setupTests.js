import * as enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import moment from 'moment'
import momentTimezone from 'moment-timezone'
import { init as initFeatureToggles } from '../util/featureToggles'

enzyme.configure({ adapter: new Adapter() })

moment.locale('de')
momentTimezone.locale('de')

initFeatureToggles()
