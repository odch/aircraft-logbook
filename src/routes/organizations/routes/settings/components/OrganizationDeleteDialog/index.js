import OrganizationDeleteDialog from './OrganizationDeleteDialog'

export default OrganizationDeleteDialog
