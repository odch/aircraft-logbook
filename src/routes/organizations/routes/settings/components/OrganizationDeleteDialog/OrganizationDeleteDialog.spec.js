import React from 'react'
import { renderIntlMaterial } from '../../../../../../testutil/renderIntl'
import OrganizationDeleteDialog from './OrganizationDeleteDialog'

describe('components', () => {
  describe('OrganizationDeleteDialog', () => {
    it('renders correctly', () => {
      const renderedValue = renderIntlMaterial(
        <OrganizationDeleteDialog
          data={{ name: 'my_org' }}
          updateData={() => {}}
          open
        />
      )
      expect(renderedValue.html()).toMatchSnapshot()
    })
  })
})
