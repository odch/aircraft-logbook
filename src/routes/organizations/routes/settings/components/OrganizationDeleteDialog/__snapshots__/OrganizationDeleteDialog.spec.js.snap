// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes settings components OrganizationDeleteDialog renders correctly 1`] = `"<div class=\\"MuiModal-root-13 MuiDialog-root-1 MuiDialog-scrollPaper-2\\" role=\\"dialog\\" data-cy=\\"organization-delete-dialog\\"><div class=\\"MuiBackdrop-root-15\\" aria-hidden=\\"true\\" style=\\"opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div class=\\"MuiPaper-root-17 MuiPaper-elevation24-43 MuiPaper-rounded-18 MuiDialog-paper-4 MuiDialog-paperScrollPaper-5 MuiDialog-paperWidthSm-8\\" style=\\"opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"document\\" tabindex=\\"-1\\"><div class=\\"MuiDialogTitle-root-44\\"><h2 class=\\"MuiTypography-root-45 MuiTypography-title-51\\"><span>Organisation löschen</span></h2></div><form><div class=\\"MuiDialogContent-root-71\\"><p class=\\"MuiTypography-root-45 MuiDialogContentText-root-72 MuiTypography-subheading-52 MuiTypography-colorTextSecondary-69\\"><span>Gib den Namen deiner Organisation ein, um die Organisation zu löschen:</span>&nbsp;<strong>my_org</strong></p><div class=\\"MuiFormControl-root-73 MuiFormControl-marginDense-75 MuiFormControl-fullWidth-76\\" data-cy=\\"organization-id-field\\"><div class=\\"MuiInputBase-root-90 MuiInput-root-77 MuiInput-underline-81 MuiInputBase-fullWidth-99 MuiInput-fullWidth-84 MuiInputBase-focused-92 MuiInput-focused-79 MuiInputBase-formControl-91 MuiInput-formControl-78 MuiInputBase-marginDense-97\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-100 MuiInput-input-85 MuiInputBase-inputMarginDense-101 MuiInput-inputMarginDense-86\\" id=\\"name\\" placeholder=\\"Name\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div></div></div><div class=\\"MuiDialogActions-root-107\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-135 MuiButton-root-109 MuiButton-text-111 MuiButton-flat-114 MuiDialogActions-action-108\\" type=\\"button\\"><span class=\\"MuiButton-label-110\\"><span>Abbrechen</span></span><span class=\\"MuiTouchRipple-root-138\\"></span></button><button tabindex=\\"-1\\" class=\\"MuiButtonBase-root-135 MuiButtonBase-disabled-136 MuiButton-root-109 MuiButton-contained-120 MuiButton-containedSecondary-122 MuiButton-raised-123 MuiButton-raisedSecondary-125 MuiButton-disabled-129 DeleteButton-button-145\\" type=\\"submit\\" disabled=\\"\\" data-cy=\\"organization-delete-button\\"><span class=\\"MuiButton-label-110\\"><svg class=\\"MuiSvgIcon-root-147 DeleteButton-leftIcon-146\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg>Organisation löschen</span></button></div></form></div></div>"`;
